src/Orchard.Web/Modules/Orchard.OpenId/InMemoryCache.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+namespace Orchard.OpenId {
+    public class InMemoryCache : TokenCache {
```

Comment:
Just have this class implement `ISingletonDependency` and then resolve it instead of instantiating it.